code/modules/mob/living/carbon/human/species.dm

```diff
@@ -185,6 +191,11 @@
 
 /datum/species/proc/handle_hair(mob/living/carbon/human/H, forced_colour)
 	H.remove_overlay(HAIR_LAYER)
+
+	var/obj/item/bodypart/head/HD = H.get_bodypart("head")
+	if(!HD || HD.status == ORGAN_ROBOTIC) //Decapitated or robotic head
```

Comment:
The issue is that replacing a decapitated head with a robotic one use the same code as head augmenting, so you can't have hair with augments but not with robotic prosthetic head.